src/main/java/net/helpscout/api/extractors/HashExtractor.java

```diff
@@ -0,0 +1,36 @@
+package net.helpscout.api.extractors;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+import net.helpscout.api.ResultExtractor;
+import net.helpscout.api.utils.HTTPConnectionUtils;
+import org.slf4j.LoggerFactory;
+
+import java.net.HttpURLConnection;
+
+/**
+ * @Author: ivan
+ * Date: 20.05.16
+ * Time: 1:16
+ */
+public class HashExtractor implements ResultExtractor<String> {
+
+    public String extract(HttpURLConnection conn) {
+        String hash = null;
+        String response;
+        try {
+            response = HTTPConnectionUtils.getResponse(conn);
+            LoggerFactory.getLogger(getClass()).debug("attachment: {}",
+                    response);
+            JsonElement obj = (new JsonParser()).parse(response);
+            JsonElement item = obj.getAsJsonObject().get("item");
+            hash = item.getAsJsonObject().get("hash").getAsString();
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
Actually, the main idea with Extractors just move it out from ApiClient class, so I copy this code to separate classes, just looking on them briefly.
As for this issue, I agree it is bad practice and it worth to be fixed. I'm not sure that refactoring of these files (Extractors) should go in this PR, cause the main idea here - make ApiCLient lightweight class.